edalize/tools/vpr.py

```diff
@@ -88,6 +88,9 @@ def configure(self, edam):
             if f.file_type in ["SDC"]:
                 timing_constraints.append(f.name)
 
+        if self.tool_options.get("input_file", "") != "":
+            file_netlist += [self.tool_options.get("input_file", "")];
+
```

Comment:
@olofk I came up with this better fix than the file_type method, I think I can do better but I need your help because I don't understand how that src_files array is populated (It's empty the way F4PGA uses the VPR tool)

Reply:
So... I'm looking at the vpr tool class now and that's not how it's supposed to work. I must have been incredibly sloppy when I reviewied that PR. I wonder if it's easier if I try to fix it first or if you do that as part of your PR. You're doing most of the fixing in your PR already so it might be best if you just fix the last part as well. The problem now is that the code loops over `src_files` which you correctly noted is empty. The correct way is to loop over `self.files`, pick out the files of interest (e.g. blif/eblif and sdc files in this case) and then store all unused files in a list that can be added to the output edam file so that any tool that might follow vpr in a flow get passed all files that vpr didn't consume. I think you can look at pretty much any tool class (except vpr) and see what the `for f in self.files:` loop looks like

Reply:
I made a patch [here](https://github.com/olofk/edalize/tree/fixvpr) that you can use for inspiration if you want to fix the vpr class through your PR